pkg/adaptor/cloud/aws/provider.go

```diff
@@ -98,6 +98,22 @@ func (p *awsProvider) CreateInstance(ctx context.Context, podName, sandboxID str
 	//Convert userData to base64
 	userDataEnc := base64.StdEncoding.EncodeToString([]byte(userData))
 
+	logger.Printf("instance type requested %q", instanceType)
+
+	// If instanceType is empty, set instanctType to default. Non-empty instanceType is set via annotation
+	// Or
+	// If instanceTypes is empty and instanceType is not default, set instanceType to default
+	// Or
+	// If instanceTypes is not empty and instanceType is not among the supported instance types, set instanceType to default
+	if instanceType == "" ||
```

Comment:
I would factor this out into a function and add a unit test to spec the desired behaviour

Reply:
How about if I make this a generic utility method that can be called by any provider?